### DISCUSSION TITLE
How to extract tabular data with no clear table format

### DISCUSSION THREAD

**Discussion by Anasuya-pautra** [jsvine/pdfplumber#763] [Ask for help with specific PDFs]
[WIMEQ288.pdf](https://github.com/jsvine/pdfplumber/files/10059086/WIMEQ288.pdf)

Hi,

please check the attached PDF,.

I want to extract tabular data "Composition","Top 10 Holdings","Morningstar Equity Sectors" and also Market cap which has no header.
it it possible to extract the data in excel file and repeat this process for multiple similar files?

Thank you for your valuable time and help in advance.

**Comment by jsvine** [comment]
Hi @Anasuya-pautra: Yes, this is possible — but it will require you to write custom logic to identify the information. For tasks like these, you may find methods such as `Page.extract_words(...)` helpful, as well as `Page.rects`, `Page.lines`, and `Page.curves`.

**Reply by Anasuya-pautra** [reply]
Thank you for commenting

**Comment by Anasuya-pautra** [comment]
Can anyone help me with this? I am not able to resolve this

**Reply by samkit-jain** [reply]
Could you please share what all you have tried? As Jeremy, mentioned, you would need to apply your custom logic. It could mean cropping the page, using different table extraction settings and having pre/post-processing
